system/mp/mpcomm.cpp

```diff
@@ -762,22 +885,38 @@ protected: friend class CMPPacketReader;
                     throw e;
                 }
 #ifdef _TRACE
+                StringBuffer str;
                 str.clear();
                 LOG(MCdebugInfo(100), unknownJob, "MP: Retrying connection to %s, %d attempts left",remoteep.getUrlStr(str).toCharArray(),retrycount+1);
 #endif
             }
+
             ::Release(newsock);
             newsock = NULL;
+
             {
                 CriticalUnblock unblock(connectsect); // to avoid connecting philosopher problem
 #ifdef _FULLTRACE
-                    PROGLOG("MP: before sleep");
+                PROGLOG("MP: before sleep");
 #endif
-                Sleep(2000+getRandom()%3000);
+                 // Sleep(2000+getRandom()%3000);
```

Comment:
Instead of this old line, a comment describing why a short sleep loop helps would be better.